bleak/backends/scanner.py

```diff
@@ -41,3 +44,18 @@ async def set_scanning_filter(self, **kwargs):
     @abc.abstractmethod
     async def get_discovered_devices(self) -> List[BLEDevice]:
         raise NotImplementedError()
+
+    @classmethod
+    @abc.abstractmethod
+    async def find_specific_device(cls, device_identifier: str, timeout: float = 10.0) -> BLEDevice:
```

Comment:
How about calling this `find_device_by_id()`? This way we could have other methods like find device with service UUID or find device with manufacturer specific data?

Reply:
I thought a very shrt while on naming for this. Picked one and moved forward, meant to change it at the end to something more appropriate.

Yes, you are right. I meant those filtering methods to be unimplemented in Bleak and left to users to implement by themselves, to avoid having to support the code. But the name is still better than `find_specific_device`. 

`find_device_by_address`?

Reply:
> find_device_by_address

I like this name too.